cloud/services/virtualmachines/virtualmachines.go

```diff
@@ -79,7 +80,11 @@ func (s *Service) Reconcile(ctx context.Context) error {
 
 	vmSpec := s.Scope.VMSpec()
 	existingVM, err := s.getExisting(ctx, vmSpec.Name)
+
 	switch {
+	case err != nil && azure.ResourceNotFound(err) && s.Scope.ProviderID() != "":
+		s.Scope.SetVMState("")
```

Comment:
Should we have a new state for deleted vms?

Reply:
the states are based on https://docs.microsoft.com/en-us/azure/virtual-machines/states-lifecycle#provisioning-states, but +1 on making the state explicit in this case since there isn't really an equivalent.